providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/api/GoFeatureFlagApi.java

```diff
@@ -0,0 +1,283 @@
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * GoFeatureFlagApi is the class to contact the GO Feature Flag relay proxy.
+ */
+@Slf4j
+public final class GoFeatureFlagApi {
+    /** apiKey contains the token to use while calling GO Feature Flag relay proxy. */
+    private final String apiKey;
+
+    /** httpClient is the instance of the OkHttpClient used by the provider. */
+    private final OkHttpClient httpClient;
+
+    /** parsedEndpoint is the endpoint of the GO Feature Flag relay proxy. */
+    private final HttpUrl parsedEndpoint;
+
+    /**
+     * GoFeatureFlagController is the constructor of the controller to contact the GO Feature Flag
+     * relay proxy.
+     *
+     * @param options - options to initialise the controller
+     * @throws InvalidOptions - if the options are invalid
+     */
+    @Builder
+    private GoFeatureFlagApi(final GoFeatureFlagProviderOptions options) throws InvalidOptions {
+        Validator.providerOptions(options);
+        this.apiKey = options.getApiKey();
+        this.parsedEndpoint = HttpUrl.parse(options.getEndpoint());
+        if (this.parsedEndpoint == null) {
+            throw new InvalidEndpoint();
+        }
+
+        // Register JavaTimeModule to be able to deserialized java.time.Instant Object
+        Const.SERIALIZE_OBJECT_MAPPER.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+        Const.SERIALIZE_OBJECT_MAPPER.enable(SerializationFeature.INDENT_OUTPUT);
+        Const.SERIALIZE_OBJECT_MAPPER.registerModule(new JavaTimeModule());
+
+        int timeout = options.getTimeout() == 0 ? 10000 : options.getTimeout();
+        long keepAliveDuration = options.getKeepAliveDuration() == null ? 7200000 : options.getKeepAliveDuration();
+        int maxIdleConnections = options.getMaxIdleConnections() == 0 ? 1000 : options.getMaxIdleConnections();
+
+        this.httpClient = new OkHttpClient.Builder()
+                .connectTimeout(timeout, TimeUnit.MILLISECONDS)
+                .readTimeout(timeout, TimeUnit.MILLISECONDS)
+                .callTimeout(timeout, TimeUnit.MILLISECONDS)
+                .readTimeout(timeout, TimeUnit.MILLISECONDS)
+                .writeTimeout(timeout, TimeUnit.MILLISECONDS)
+                .connectionPool(new ConnectionPool(maxIdleConnections, keepAliveDuration, TimeUnit.MILLISECONDS))
+                .build();
+    }
+
+    /**
+     * evaluateFlag is calling the GO Feature Flag relay proxy to evaluate the feature flag.
+     *
+     * @param key               - name of the flag
+     * @param evaluationContext - context of the evaluation
+     * @return EvaluationResponse with the evaluation of the flag
+     * @throws OpenFeatureError - if an error occurred while evaluating the flag
+     */
+    public GoFeatureFlagResponse evaluateFlag(final String key, final EvaluationContext evaluationContext)
```

Comment:
why is this option on this provider ? isn't the generic OFREP doing something similar ?